src/wallet/wallet.cpp

```diff
@@ -4032,20 +4052,27 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
 
     if (fFirstRun)
     {
-        // ensure this wallet.dat can only be opened by clients supporting HD with chain split and expects no default key
-        walletInstance->SetMinVersion(FEATURE_LATEST);
+        // ensure this wallet can only be opened by clients supporting HD with chain split and expects no default key
+        if (!create_empty) {
+            walletInstance->SetMinVersion(FEATURE_PRE_SPLIT_KEYPOOL);
+        } else {
+            // In addition, this wallet can only be opened by clients supporting empty wallets
+            walletInstance->SetMinVersion(FEATURE_LATEST);
+        }
 
         if ((wallet_creation_flags & WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
             //selective allow to set flags
             walletInstance->SetWalletFlag(WALLET_FLAG_DISABLE_PRIVATE_KEYS);
+        } else if (create_empty) {
+            // Do not generate a new seed
         } else {
```

Comment:
Instead of an empty else if block, change the else to be `else if (!create_empty)`.